Source/NSGridView.m

```diff
@@ -0,0 +1,957 @@
+          if ([coder containsValueForKey: @"NSGrid_xPlacement"])
+            {
+              _xPlacement = [coder decodeIntegerForKey: @"NSGrid_xPlacement"];
+            }      
+          if ([coder containsValueForKey: @"NSGrid_yPlacement"])
+            {
+              _yPlacement = [coder decodeIntegerForKey: @"NSGrid_yPlacement"];
+            }  
+          if ([coder containsValueForKey: @"NSGrid_alignment"])
+            {
+              _rowAlignment = [coder decodeIntegerForKey: @"NSGrid_alignment"];
+            }      
+        }
+      else
+        {
+          [self setContentView: [coder decodeObject]];
+          ASSIGN(_mergeHead, [coder decodeObject]);
+          _owningRow = [coder decodeObject]; // weak
+          [_owningRow _addCell: self];
+          _owningColumn = [coder decodeObject]; // weak
+          [_owningRow _addCell: self];
+          [coder decodeValueOfObjCType: @encode(NSInteger)
+                                    at: &_xPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_yPlacement];
+          [coder decodeValueOfObjCType:@encode(NSInteger)
+                                    at: &_rowAlignment];
+        }
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_mergeHead);
+  [super dealloc];
+}
+
+@end
+
+/// Column ///
+@implementation NSGridColumn
+
++ (void) initialize
+{
+  if (self == [NSGridColumn class])
+    {
+      [self setVersion: 1];
+    }
+}
+
+- (void) _addCell: (NSGridCell *)c
+{
+  [_cells addObject: c];
+}
+
+- (NSGridView *) gridView
+{
+  return _gridView;
+}
+
+- (NSInteger) numberOfCells
+{
+  return [_cells count];
+}
+
+- (NSGridCell *) cellAtIndex:(NSInteger)index
+{
+  return [_cells objectAtIndex: index];
+}
+
+- (NSGridCellPlacement) xPlacement
+{
+  return _xPlacement;
+}
+
+- (void) setXPlacement: (NSGridCellPlacement)x
+{
+  _xPlacement = x;
+}
+
+- (CGFloat) width
+{
+  return _width;
+}
+
+- (void) setWidth: (CGFloat)f
+{
+  _width = f;
+}
+
+- (CGFloat) leadingPadding
+{
+  return _leadingPadding;
+}
+
+- (void) setLeadingPadding: (CGFloat)f
+{
+  _leadingPadding = f;
+}
+
+- (CGFloat) trailingPadding
+{
+  return _trailingPadding;
+}
+
+- (void) setTrailingPadding: (CGFloat)f
+{
+  _trailingPadding = f;
+}
+
+- (BOOL) isHidden
+{
+  return _isHidden;
+}
+
+- (void) mergeCellsInRange: (NSRange)range
+{
+}
+
+// coding
+- (void) encodeWithCoder: (NSCoder *)coder
+{
+  if ([coder allowsKeyedCoding])
+    {
+      [coder encodeBool: _isHidden forKey: @"NSGrid_hidden"];
+      [coder encodeFloat: _leadingPadding forKey: @"NSGrid_leadingPadding"];
+      [coder encodeObject: _gridView forKey: @"NSGrid_owningGrid"]; // weak
+      [coder encodeFloat: _trailingPadding forKey: @"NSGrid_trailingPadding"];
+      [coder encodeFloat: _width forKey: @"NSGrid_width"];
+      [coder encodeInteger: _xPlacement forKey: @"NSGrid_xPlacement"];
+    }
+  else
+    {
+      [coder encodeValueOfObjCType:@encode(BOOL)
```

Comment:
Could you please add spaces after the colons to all these methods?

Reply:
> This PR should stillte labeled as draft as it is not functional in its current state.

Okay.